### ISSUE TITLE
Indent not working at ckeditor5-build-classic

### ISSUE THREAD

**Issue by MahediSabuj** [ckeditor/ckeditor5#5658]
## 📝 Provide detailed reproduction steps (if any)

1. Clone [ckeditor5-build-classic](https://github.com/ckeditor/ckeditor5-build-classic) from GitHub repository
2. Open sample/index.html
3. 'Indent' toolbar not activated when I put the cursor on **Sample**.

### ✔️ Expected result

![image](https://user-images.githubusercontent.com/35859676/67517746-10a8d100-f6c5-11e9-8351-8cd8cd05b2ce.png)

### ❌ Actual result

![image](https://user-images.githubusercontent.com/35859676/67517530-9b3d0080-f6c4-11e9-8130-f186fd335a15.png)

## 📃 Other details

* Browser: Google Chrome
* OS: Windows 10
* CKEditor version: 15.0.0
* Installed CKEditor plugins: …

---

After looking into indentation [docs](https://ckeditor.com/docs/ckeditor5/latest/features/indent.html), I updated `app.js` with the following snippet, then it works as expected.

```javascript
// Other Imports
import IndentBlock from '@ckeditor/ckeditor5-indent/src/indentblock';

ClassicEditor.builtinPlugins = [
  // Other Plugins
  IndentBlock 
];
````
---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by mlewand** [comment]
This is actually by design. There are 2 implementations for indent plugin: one for lists (only) and the other for text blocks.

We decided to enable both implementations **only** for the [document editor](https://ckeditor.com/docs/ckeditor5/latest/examples/builds/document-editor.html) build, while **all** other builds have only list indentation implementation. Meaning that indenting block is unavailable.

The proper solution is add IndentBlock plugin just like you have proposed.

It's early after the release, I'll leave this ticket open for a while to ensure whether this is not confusing for more users.

**Comment by peppies** [comment]
The proper solution is to be able to indent a simple paragraph by default... This isn't worthy of a plugin, it is worthy of basic English writing format...

Definition of "indent" - to set in or back from the margin, as the first line of a PARAGRAPH.
https://www.dictionary.com/browse/indent

**Comment by Thanujabadabagni** [comment]
indents is not working even in nov 2020

**Comment by Reinmar** [comment]
I gratefully accept the rebuke.

The default builds of CKEditor 5 are focused on semantic content. Using indentation is not semantical – it's purely presentational. Also, you actually confused line and block indentation. This ticket is about block indentation, while you refer to first line indentation.

Perhaps someone may correct me, but the latter (first line indentation) is exactly the thing that should be achieved with stylesheets because it's a matter of style. And that'd be true for both in the web and in web processors. And you can do that too – just add `p { text-indent: 1em }` to your stylesheet.

A side note about first line indentation – while it's common in print, it's not common in web typography.

Finally, I understand that there will be many people who will miss the indent feature in the default builds (well, 4 of them, because it's present in the document build). But:

*   Same thing will be true for fonts (colors, sizes, family), alignment, horizontal lines and many others. They are all popular features. Should we add them all?
*   The editor is configurable and everyone can adjust it to their needs. While I understand it may be work one wants to avoid, you'll likely have to add some other features yourself anyway. We may try to have sane defaults, but there will be usually something missing for a particular case.

**Comment by Subhra264** [comment]
@Reinmar Sorry, I am little confused if it is a `ckeditor5-react` issue or a `ckeditor5-build-classic` issue, but as my issue is somewhat similar to this one, so I am commenting here.

I followed the react-integration [quick start](https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/frameworks/react.html#quick-start) guide, and everything is working fine except the lists, When I create a list, it looks like the following ---
 
![ck-editor_issue](https://user-images.githubusercontent.com/59690052/125471017-98c7402b-b439-4020-86df-d8fa33240378.png)

That is, it is creating an ordered list, but the digits get cut off. Similarly for unordered lists, the bullet markers can not be even seen. 

![ck-editor_bullet_list_issue](https://user-images.githubusercontent.com/59690052/125472219-727d806c-98ce-4da9-9e00-63f73a133b23.png)

Neither do nested lists work. But I could not find any related fix for it as if no one else has encountered this problem. That's why I am confused, here is my code ( totally copied from the documentation ) ---

```jsx
import { CKEditor } from '@ckeditor/ckeditor5-react';
import ClassicEditor from '@ckeditor/ckeditor5-build-classic';

export default function Editor (props) {

  return (
    <div className="editor-container">
      <CKEditor
        editor={ ClassicEditor }
        data="<p>Hello from CKEditor 5!</p>"
        onReady={ editor => {
          // You can store the "editor" and use when it is needed.
          console.log( 'Editor is ready to use!', editor );
        }}
        onChange={( event, editor ) => {
          const data = editor.getData();
          console.log( { event, editor, data } );
        }}
        onBlur={( event, editor ) => {
          console.log( 'Blur.', editor );
        }}
        onFocus={( event, editor ) => {
          console.log( 'Focus.', editor );
        }}
      />
    </div>
  );
}
```

Thanks in advance for any help...

**Comment by coderrohanpahwa** [comment]
How to get working the indent feature with vanilla javascript?

**Comment by evanness** [comment]
I've not been able to get indent to work in React. It doesn't just work with the buttons, which seems like a really bad design. I've not been able to import the Indent or IndentBlock plugins without getting the `CKEditorError: ckeditor-duplicated-modules` error.

**Comment by FilipTokarski** [comment]
@evanness That's most probably because you are trying to add a plugin to a predefined build, which will not work. Please check the [docs on customising editor in React](https://ckeditor.com/docs/ckeditor5/latest/installation/getting-started/frameworks/react.html#customizing-the-builds).

**Comment by evanness** [comment]
@FilipTokarski Thanks for the link. I had already explored with that no luck. I've done custom builds with and without the online builder and I keep running into issues. I tried many options including customizing my webpack as suggested (and with other variations). I've decided that Quill is the better option for me. It works with very little setup.

**Comment by azizrosyid** [comment]
if you use nextjs and have styling 
`
* {
padding: 0;
margin: 0;
}
`

remove this from your global css

**Comment by maldimz** [comment]
Thanks man, it works for me, i use nextjs and CkEditor Classic

Before
![image](https://github.com/ckeditor/ckeditor5/assets/79178557/4aa11cb8-cc1b-4dfa-84c0-d72cdf5a921f)

![image](https://github.com/ckeditor/ckeditor5/assets/79178557/d2b915ae-4d7e-41d6-8e11-0463b6059e1c)

After
![image](https://github.com/ckeditor/ckeditor5/assets/79178557/b2738b85-9e64-438a-8133-403d408b174d)

**Comment by Witoso** [comment]
Hello everyone, happy to report that we finished the looong work in [#15502](https://github.com/ckeditor/ckeditor5/issues/15502), and the new installation methods shipped in [v42.0.0](https://github.com/ckeditor/ckeditor5/releases/tag/v42.0.0) allow seamless dynamic extension of the editor's plugins in npm and CDN setups.

Predefined builds will still be released in new versions for some time, but they are put on the deprecation path and we don't plan to update them.

Test the new setup ([migration docs](https://ckeditor.com/docs/ckeditor5/latest/updating/nim-migration/migration-to-new-installation-methods.html) and [quick start](https://ckeditor.com/docs/ckeditor5/latest/getting-started/installation/quick-start.html)), and let us know what you think!
